### PR TITLE
Like video

### PR DESCRIPTION
**Pull request by gopalbbharadva**
#### Features:
- toggle between like, dislike
- navigate to single video page from like and watch later page.
- Remove from like video list on like_video page.

Live Preview: https://deploy-preview-8--vframe.netlify.app/

### CODE REVIEW

File: src/frontend/pages/watchlater/WatchLaterPage.jsx [hunk 1]
```diff
@@ -24,23 +25,28 @@ export const WatchLaterPage = () => {
...
             />
             <p className="fs-xlg mg-vrtl-md">Watch later videos</p>
             <p className="fs-md mg-vrtl-sm">
               {watchLaterVideos.length}
               {watchLaterVideos.length <= 1 ? " Video" : " Videos"}
             </p>
           </div>
-          <div className="watch-later-videos">
+          <div className="fv-videos">
             {watchLaterVideos.length >= 1 ? (
```
Comment (BelideNikhil) [inline src/frontend/pages/watchlater/WatchLaterPage.jsx:41]: you can use optional chaining here

**Review by BelideNikhil — APPROVED** [review]
The UI good and the code is clean as well
One suggestion is that when a user logs out, the liked state still remains on the video and it does not redirect to login page.

**Comment by gopalbbharadva** [comment]
> The UI good and the code is clean as well One suggestion is that when a user logs out, the liked state still remains on the video and it does not redirect to login page.

Ok i am fixing.

File: src/frontend/helperfunctions/likeHandler.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+} from "../services/likeService";
+
+export const getLikeHandler = async (token, dataStoreDispatch) => {
+  try {
+    const res = await getLikeService(token);
+    console.log('like res',res.data.likes)
+    if (res.status === 200) {
+      dataStoreDispatch({
+        type: actionTypes.LIKE_VIDEO,
+        paylaod: res.data.likes,
```
Comment (MarkVed17) [inline src/frontend/helperfunctions/likeHandler.js]: payload*


nice catch, updating

File: src/frontend/components/HorizontalVideoCard/HorizontalVideoCard.jsx [hunk 3]
```diff
@@ -1,32 +1,34 @@
 import React from "react";
 import { FaTrashAlt } from "react-icons/fa";
+import { Link } from "react-router-dom";
 import { useAuth } from "../../contexts/AuthContext";
 import { useDataStore } from "../../contexts/DataStoreContext";
```
Comment (mimansha-swarup) [inline src/frontend/components/HorizontalVideoCard/HorizontalVideoCard.jsx]: can have barrel export for useAuth and useDataStore
you can do it on next go

**Review by MarkVed17 — APPROVED** [review]
I like the naming practices you followed. Code becomes much simpler to read. You may proceed.

**Review by mimansha-swarup — APPROVED** [review]
code looks clean 
good to merge

**Pull request merged**